Upgraded dependencies

remojansen opened the pull request.

remojansen commented:
Hi @lholznagel any ideas about why after all tests past, [the CI build times out](https://travis-ci.org/inversify/inversify-express-example/jobs/297364915#L1051-L1052)?

> No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

lholznagel commented:
Nope, travis is the same.

Will take a look at it

lholznagel commented:
As soon as I start a mongodb tests, it is not terminating. Going back to version 2.2.31 the problem still appears.

lholznagel commented:
Closing the db after everything is done also seems not to fix it :(

The only hack I found was to call `process.exit(0)` after all tests are done. But this is very very dirty.

remojansen commented:
@lholznagel thanks for trying it. I will keep researching I don't want to call `process.exit(0)`. Will try to Google it maybe someone is had this issue already.

remojansen commented:
Hi @lholznagel I found a way to fix this for now. It looks like the issue only takes place when we use npm scripts so I moved it to a shell script instead. 

- If `npm test` runs  `nyc -e=.ts -- mocha -t=5000 --require ts-node/register  --require reflect-metadata/Reflect **/*.spec.ts` it fails 🔥
- If `npm test` runs `./run-test.sh` it fails 🔥
- If we run `./run-test.sh` without npm scripts it works 👍 

The shell script doesn't work on AppVeyor so I removed that as well. I have reported the issue 👉 https://github.com/istanbuljs/nyc/issues/713

remojansen commented:
Seems like in CI is still failing 😢

remojansen commented:
@lholznagel it is related with https://boneskull.com/mocha-v4-nears-release/#mochawontforceexit but I added --exit and now it fails sometimes:

```
 /home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/transformer.js:97
            mappedCoverage.addStatement(mapping.loc, hits);
                           ^
TypeError: mappedCoverage.addStatement is not a function
    at /home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/transformer.js:97:28
    at Array.forEach (<anonymous>)
    at SourceMapTransformer.processFile (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/transformer.js:88:34)
    at /home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/transformer.js:169:24
    at Array.forEach (<anonymous>)
    at SourceMapTransformer.transform (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/transformer.js:159:25)
    at MapStore.transformCoverage (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/node_modules/istanbul-lib-source-maps/lib/map-store.js:143:8)
    at SourceMaps.remapCoverage (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/lib/source-maps.js:29:41)
    at NYC._getCoverageMapFromAllCoverageFiles (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/index.js:436:30)
    at NYC.report (/home/travis/build/inversify/inversify-express-example/node_modules/nyc/index.js:442:18)
```

lholznagel commented:
Okay, very interesting, ...

lholznagel commented:
We should investigate what is causing this problem. The problem seems to only occur with MongoDB tests, so there is a starting point.

remojansen commented:
I think the mongo db client is not properly closed or something like that. For now just remember that if a build fails, re-running should fix it for now.
